lib/OpenQA/Worker/Job.pm

```diff
@@ -0,0 +1,1263 @@
+        service_port_delta => 2,                     # liveviewhandler is supposed to run on web UI port + 2
+        json               => \%new_progress_info,
+        non_critical       => 1,
+        callback           => sub {
+            my ($res) = @_;
+            if (!$res) {
+                log_warning('Failed to post upload progress to liveviewhandler.');
+            }
+            $callback->($res);
+        });
+}
+
+sub upload_images {
+    my ($self) = @_;
+
+    my $tx;
+    my $job_id = $self->id;
+    my $client = $self->client;
+    my $ua     = $client->ua;
+    my $ua_url = $client->url->clone;
+    $ua_url->path("jobs/$job_id/artefact");
+
+    my $pooldir        = $self->worker->pool_directory;
+    my $fileprefix     = "$pooldir/testresults";
+    my $images_to_send = $self->images_to_send;
+    for my $md5 (keys %$images_to_send) {
+        my $file = $images_to_send->{$md5};
+        $self->_optimize_image("$fileprefix/$file");
+
+        log_debug("Uploading $file as $md5");
+        my %form = (
+            file => {
+                file     => "$fileprefix/$file",
+                filename => $file
+            },
+            image => 1,
+            thumb => 0,
+            md5   => $md5
+        );
+        # don't use api_call as it retries and does not allow form data
+        $tx = $ua->post($ua_url => form => \%form);
+        # FIXME: error handling?
+
+        $file = "$fileprefix/.thumbs/$file";
+        if (-f $file) {
+            $self->_optimize_image($file);
+            $form{file}->{file} = $file;
+            $form{thumb}        = 1;
+            $tx                 = $ua->post($ua_url => form => \%form);
+        }
+    }
+    $self->{_images_to_send} = {};
+
+    for my $file (@{$self->files_to_send}) {
+        log_debug("Uploading $file");
+        my %form = (
+            file => {
+                file     => "$pooldir/testresults/$file",
+                filename => $file,
+            },
+            image => 0,
+            thumb => 0,
+        );
+        # don't use api_call as it retries and does not allow form data
+        # (refactor at some point)
+        $tx = $ua->post($ua_url => form => \%form);
+        # FIXME: error handling?
+    }
+    $self->{_files_to_send} = [];
+    return !$tx || !$tx->error;
+}
+
+sub _upload_log_file_or_asset {
+    my ($self, $upload_parameter) = @_;
+
+    my $filename = $upload_parameter->{file}->{filename};
+    my $file     = $upload_parameter->{file}->{file};
+    my $is_asset = $upload_parameter->{asset};
+    log_info("Uploading $filename", channels => ['worker', 'autoinst'], default => 1);
+
+    if ($is_asset) {
+        return $self->_upload_asset($upload_parameter);
+    }
+    else {
+        return $self->_upload_log_file($upload_parameter);
+    }
+}
+
+sub _log_upload_error {
+    my ($filename, $res) = @_;
+
+    my $err = $res->error;
+    return undef unless $err;
```

Comment:
It's already a postfix conditional. 🤣 Just make it `return undef unless my $err = $res->error;`.